model/command_response.go

```diff
@@ -32,31 +33,30 @@ type CommandResponse struct {
 }
 
 func CommandResponseFromHTTPBody(contentType string, body io.Reader) (*CommandResponse, error) {
-	if strings.TrimSpace(strings.Split(contentType, ";")[0]) == "application/json" {
-		return CommandResponseFromJSON(body)
-	}
-	if b, err := io.ReadAll(body); err == nil {
-		return CommandResponseFromPlainText(string(b)), nil
+	if bodyRaw, err := io.ReadAll(body); err != nil {
+		return nil, nil
+	} else if commandResponse, err := CommandResponseFromJSON(bodyRaw); err == nil {
+		return commandResponse, nil
+	} else if ContentTypeIs(contentType, "application/json") {
+		return nil, err
+	} else {
+		return &CommandResponse{
+			Text: string(bodyRaw),
+		}, nil
 	}
-	return nil, nil
 }
 
-func CommandResponseFromPlainText(text string) *CommandResponse {
-	return &CommandResponse{
-		Text: text,
-	}
+func ContentTypeIs(rawContentType string, expectedContentType string) bool {
+	givenContentType := strings.TrimSpace(strings.Split(rawContentType, ";")[0])
+	return givenContentType == expectedContentType
```

Comment:
We should make sure the return value of `Split` is at least 1 element length before accessing thr `0` index. I know it's basically impossible for this not to be the case, but index access should always be guarded by some conditional check first. Otherwise the plugin will panic if the array is empty

Reply:
Good point, I'll work on that